validator/keymanager/remote-web3signer/keymanager.go

```diff
@@ -0,0 +1,162 @@
+	}
+	if cfg.PublicKeysURL == "" && len(cfg.ProvidedPublicKeys) == 0 {
+		return nil, errors.New("no valid public key options provided")
+	}
+	client, err := newClient(cfg.BaseEndpoint)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create client")
+	}
+	return &Keymanager{
+		client:                Web3SignerClient(client),
+		genesisValidatorsRoot: cfg.GenesisValidatorsRoot,
+		accountsChangedFeed:   new(event.Feed),
+		publicKeysURL:         cfg.PublicKeysURL,
+		providedPublicKeys:    cfg.ProvidedPublicKeys,
+	}, nil
+}
+
+// FetchValidatingPublicKeys fetches the validating public keys from the remote server or from the provided keys.
+func (km *Keymanager) FetchValidatingPublicKeys(_ context.Context) ([][48]byte, error) {
+	if km.publicKeysURL != "" {
+		providedPublicKeys, err := km.client.GetPublicKeys(km.publicKeysURL)
+		if err != nil {
+			return nil, err
+		}
+		km.providedPublicKeys = providedPublicKeys
+	}
+	return km.providedPublicKeys, nil
+}
+
+// Sign signs the message by using a remote web3signer server.
+func (km *Keymanager) Sign(_ context.Context, request *validatorpb.SignRequest) (bls.Signature, error) {
+
+	if request.Fork == nil {
+		return nil, errors.New("invalid sign request: Fork is nil")
+	}
+
+	signRequestType, err := getSignRequestType(request)
+	if err != nil {
+		return nil, err
+	}
+
+	forkData := &Fork{
+		PreviousVersion: hexutil.Encode(request.Fork.PreviousVersion),
+		CurrentVersion:  hexutil.Encode(request.Fork.CurrentVersion),
+		Epoch:           fmt.Sprint(request.Fork.Epoch),
+	}
+	forkInfoData := &ForkInfo{
+		Fork:                  forkData,
+		GenesisValidatorsRoot: hexutil.Encode(km.genesisValidatorsRoot),
+	}
+	aggregationSlotData := &AggregationSlot{Slot: fmt.Sprint(request.AggregationSlot)}
+	web3SignerRequest := SignRequest{
+		Type:            signRequestType,
+		ForkInfo:        forkInfoData,
+		SigningRoot:     hexutil.Encode(request.SigningRoot),
+		AggregationSlot: aggregationSlotData,
+	}
+	return km.client.Sign(hexutil.Encode(request.PublicKey), &web3SignerRequest)
+}
+
+// getSignRequestType returns the type of the sign request.
+func getSignRequestType(request *validatorpb.SignRequest) (string, error) {
+	switch request.Object.(type) {
+	case *validatorpb.SignRequest_Block:
+		return "BLOCK", nil
+	case *validatorpb.SignRequest_AttestationData:
+		return "ATTESTATION", nil
+	case *validatorpb.SignRequest_AggregateAttestationAndProof:
+		return "AGGREGATE_AND_PROOF", nil
+	case *validatorpb.SignRequest_Slot:
+		return "AGGREGATION_SLOT", nil
+	case *validatorpb.SignRequest_BlockV2:
+		return "BLOCK_V2", nil
+	// Not supported in web3signer yet we need to add it once the merge is scheduled.
+	/*
+		case *validatorpb.SignRequest_BlockV3:
+		return "BLOCK_V3", nil
+	*/
+
+	// We do not support "DEPOSIT" type.
+	/*
+		case *validatorpb.:
+		return "DEPOSIT", nil
+	*/
```

Comment:
How will we remember to do this? This is easy to forget. Maybe we can have an open issue and then add the comment
```
// TODO(#10020): Need to add support for merge blocks
```